File: java-to-kotlin-bookmanager/src/main/java/com/example/javatokotlinbookmanager/controller/book/BookController.java
package com.example.javatokotlinbookmanager.controller.book;

import com.example.javatokotlinbookmanager.dto.book.request.BookLoanRequest;
import com.example.javatokotlinbookmanager.dto.book.request.BookRequest;
import com.example.javatokotlinbookmanager.dto.book.request.BookReturnRequest;
import com.example.javatokotlinbookmanager.service.book.BookService;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class BookController {

  private final BookService bookService;

  public BookController(BookService bookService) {
    this.bookService = bookService;
  }

  @PostMapping("/book")
  public void saveBook(@RequestBody BookRequest request) {
    bookService.saveBook(request);
  }

  @PostMapping("/book/loan")
  public void loanBook(@RequestBody BookLoanRequest request) {
    bookService.loanBook(request);
  }

  @PutMapping("/book/return")
  public void returnBook(@RequestBody BookReturnRequest request) {
    bookService.returnBook(request);
  }

}

File: java-to-kotlin-bookmanager/src/main/kotlin/com/example/javatokotlinbookmanager/controller/book/BookController.kt
package com.example.javatokotlinbookmanager.controller.book

import com.example.javatokotlinbookmanager.dto.book.request.BookLoanRequest
import com.example.javatokotlinbookmanager.dto.book.request.BookRequest
import com.example.javatokotlinbookmanager.dto.book.request.BookReturnRequest
import com.example.javatokotlinbookmanager.dto.book.response.BookStatResponse
import com.example.javatokotlinbookmanager.service.book.BookService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class BookController @Autowired constructor(
    private val bookService: BookService
) {

    @PostMapping("/book")
    fun saveBook(@RequestBody request: BookRequest) {
        bookService.saveBook(request)
    }

    @PostMapping("/book/loan")
    fun loanBook(@RequestBody request: BookLoanRequest) {
        bookService.loanBook(request)
    }

    @PutMapping("/book/return")
    fun returnBook(@RequestBody request: BookReturnRequest): Unit {
        bookService.returnBook(request)
    }

    @GetMapping("/book/loan")
    fun countLoadedBook(): Int {
        return bookService.countLoadedBook()
    }

    @GetMapping("/book/stat")
    fun getBookStatistics(): List<BookStatResponse> {
        return bookService.getBookStatistics()
    }
}